models/decision_phantom.go

```diff
@@ -1,9 +1,39 @@
 package models
 
+import (
+	"time"
+
+	"github.com/google/uuid"
+)
+
 type CreatePhantomDecisionInput struct {
 	OrganizationId     string
 	Scenario           Scenario
 	ClientObject       ClientObject
 	Pivot              *Pivot
 	TriggerObjectTable string
 }
+
+type PhantomDecision struct {
+	PhantomDecisionId   string
+	CreatedAt           time.Time
+	OrganizationId      string
+	Outcome             Outcome
+	ScenarioId          string
+	ScenarioIterationId string
+	Score               int
+	RuleExecutions      []RuleExecution
+}
+
+func AdapScenarExecToPhantomDecision(scenarioExecution ScenarioExecution) PhantomDecision {
+	return PhantomDecision{
+		PhantomDecisionId:   uuid.Must(uuid.NewV7()).String(),
+		CreatedAt:           time.Now(),
+		OrganizationId:      scenarioExecution.OrganizationId,
+		Outcome:             scenarioExecution.Outcome,
+		ScenarioId:          scenarioExecution.ScenarioId,
+		ScenarioIterationId: scenarioExecution.ScenarioIterationId,
+		Score:               scenarioExecution.Score,
+		RuleExecutions:      scenarioExecution.RuleExecutions,
```

Comment:
last thing: please add a step of "clear out the rule executions' rule_evaluation []byte field (or alternatively, create a dedicated adapter for a phantom decision rule execution model)